### ISSUE TITLE
NullReferenceException in NATS.Client.Defaults.WriteEvent()

### ISSUE THREAD

**Issue by Tridy** [nats-io/nats.net.v1#516]
src/NATS.Client/NATS.cs

```
private static void WriteEvent(String label, ConnEventArgs e) 
{
        Console.Error.WriteLine(BeginFormatMessage(label, e.Conn, null, e.Error.Message).ToString());
}
```
It looks like `e.Error` can be null and accessing `e.Error.Message` will then throw NullReferenceException

**Comment by scottf** [comment]
I will review this and track callers to see if this (and other WriteEvents, etc.)  should be guarded. Did you encounter a problem or just were reviewing the code?

**Comment by scottf** [comment]
Fixing in https://github.com/nats-io/nats.net/pull/517

**Comment by Tridy** [comment]
I encountered that when trying a sample pub-sub test:

![image](https://user-images.githubusercontent.com/7138873/142927392-464af53a-fcc1-423f-bf4d-7f135ebd7882.png)

**Comment by ColinSullivan1** [comment]
Resolved via #517, and released in [NATS.Client V 0.14.3](https://www.nuget.org/packages/NATS.Client/0.14.3).

A Regression test added with #518.

Thank you for using NATS and raising this issue!
